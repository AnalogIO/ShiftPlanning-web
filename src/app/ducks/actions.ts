import actionCreatorFactory from 'typescript-fsa';

import { LoginData } from '../types';

const actionCreator = actionCreatorFactory('shiftplanning/app');

export const login = actionCreator.async<undefined, LoginData, string>('LOGIN');

export const logout = actionCreator('LOGOUT');
