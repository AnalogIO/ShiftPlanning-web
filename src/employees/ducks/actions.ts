import { Employee, EmployeeState } from '../types';

import actionCreatorFactory from 'typescript-fsa';

const actionCreator = actionCreatorFactory('shiftplanning/employees');

export const fetchAll = actionCreator.async<undefined, EmployeeState, string>(
  'FETCH_ALL',
);

export const create = actionCreator.async<Employee, Employee, string>('CREATE');

export const update = actionCreator.async<Employee, Employee, string>('UPDATE');

export const remove = actionCreator.async<number, undefined, string>('REMOVE');
