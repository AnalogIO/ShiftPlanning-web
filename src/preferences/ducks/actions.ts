import actionCreatorFactory from 'typescript-fsa';

import { Preference } from '../types';

const actionCreator = actionCreatorFactory('shiftplanning/preferences');

export const fetchPreferences = actionCreator.async<
  number,
  { [id: number]: number },
  string
>('FETCH_PREFERENCES');

export const updatePreferences = actionCreator.async<
  number,
  { [id: number]: number },
  string
>('UPDATE_PREFERENCES');
