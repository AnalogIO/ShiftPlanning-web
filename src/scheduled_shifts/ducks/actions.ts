import actionCreatorFactory from 'typescript-fsa';

import { Schedule } from 'schedules/types';
import { ScheduledShift, ScheduledShiftState } from '../types';

const actionCreator = actionCreatorFactory('shiftplanning/scheduledShifts');

export const fetchAll = actionCreator.async<
  undefined,
  ScheduledShiftState,
  string
>('FETCH_ALL');

export const remove = actionCreator.async<
  {
    schedule: Schedule;
    scheduledShiftId: number;
  },
  undefined,
  string
>('REMOVE');

export const create = actionCreator.async<
  any, // TODO: figure out what this value is supposed to be
  {
    schedule: Schedule;
    scheduledShift: ScheduledShift;
  },
  string
>('CREATE');

export const update = actionCreator.async<
  {
    schedule: Schedule;
    scheduledShift: ScheduledShift;
  },
  undefined,
  string
>('UPDATE');
