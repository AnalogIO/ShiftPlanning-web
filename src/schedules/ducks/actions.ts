import actionCreatorFactory from 'typescript-fsa';

import { ScheduledShift } from 'scheduled_shifts/types';
import { Schedule, ScheduleDto, ScheduleState } from '../types';

type ScheduleId = number;

const actionCreator = actionCreatorFactory('shiftplanning/schedules');

export const fetchAll = actionCreator.async<undefined, ScheduleState, string>(
  'FETCH_ALL',
);

export const create = actionCreator.async<ScheduleDto, Schedule, string>(
  'CREATE',
);

export const remove = actionCreator.async<ScheduleId, number, string>('REMOVE');

export const rollout = actionCreator.async<ScheduleId, undefined, string>(
  'ROLLOUT',
);

export const generateOptimalSchedule = actionCreator.async<
  number,
  { schedule: Schedule; scheduledShifts: { [id: number]: ScheduledShift } },
  string
>('GENERATE_OPTIMAL_SCHEDULE');
