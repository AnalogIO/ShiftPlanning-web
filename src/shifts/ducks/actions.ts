import actionCreatorFactory from 'typescript-fsa';

import { Shift, ShiftState } from 'shifts/types';
import { SubmitParams } from '../NewShiftModal';

const actionCreator = actionCreatorFactory('shiftplanning/shifts');

export const fetchAll = actionCreator.async<undefined, ShiftState, string>(
  'FETCH_ALL',
);

export const create = actionCreator.async<SubmitParams, Shift, string>(
  'CREATE',
);

export const update = actionCreator.async<Shift, undefined, string>('UPDATE');

export const remove = actionCreator.async<number, undefined, string>('REMOVE');
