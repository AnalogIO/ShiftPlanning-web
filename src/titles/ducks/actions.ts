import actionCreatorFactory from 'typescript-fsa';

import { TitleState } from 'titles/types';

const actionCreator = actionCreatorFactory('shiftplanning/titles');

export const fetchAll = actionCreator.async<any, TitleState, string>(
  'FETCH_ALL',
);
