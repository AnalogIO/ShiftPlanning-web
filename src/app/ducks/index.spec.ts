import reducer, { INITIAL_STATE } from '.';
import * as actions from './actions';

describe('app', () => {
  describe('sign in', () => {
    it('starts', () => {
      const params = undefined;

      const action = actions.login.started(params);

      expect(action).toEqual({ type: 'shiftplanning/app/LOGIN_STARTED' });
      expect(reducer(INITIAL_STATE, action)).toMatchSnapshot();
    });

    it('is done', () => {
      const params = undefined;

      const result = {
        organizationName: 'Analog',
        organizationId: 1,
        token: 'abc123',
        expires: 60 * 60 * 24,
        expiresWhen: new Date(),
        employee: {
          active: true,
          checkInCount: 0,
          email: 'test@example.com',
          employeeTitle: 'Barista',
          employeeTitleId: 1,
          firstName: 'John',
          friendshipIds: [],
          id: 1,
          lastName: 'Doe',
          photoRef: '',
          roles: 0,
          wantShifts: 1,
        },
      };

      const action = actions.login.done({
        params,
        result,
      });

      expect(action).toEqual({
        type: 'shiftplanning/app/LOGIN_DONE',
        payload: {
          params,
          result,
        },
      });
      expect(reducer(INITIAL_STATE, action)).toMatchSnapshot();
    });

    it('failed', () => {
      const params = undefined;

      const error = 'Wrong username/password';

      const action = actions.login.failed({
        params,
        error,
      });

      expect(action).toEqual({
        type: 'shiftplanning/app/LOGIN_FAILED',
        error: true,
        payload: {
          params,
          error,
        },
      });
      expect(reducer(INITIAL_STATE, action)).toMatchSnapshot();
    });
  });

  describe('sign out', () => {
    it('is done', () => {
      const action = actions.logout();

      expect(action).toEqual({
        type: 'shiftplanning/app/LOGOUT',
      });
      expect(reducer(INITIAL_STATE, action)).toMatchSnapshot();
    });
  });
});
