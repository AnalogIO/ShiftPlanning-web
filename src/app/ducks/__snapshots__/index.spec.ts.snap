// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app sign in failed 1`] = `
Object {
  "currentUser": undefined,
  "loggingIn": false,
}
`;

exports[`app sign in is done 1`] = `
Object {
  "currentUser": Object {
    "active": true,
    "checkInCount": 0,
    "email": "test@example.com",
    "employeeTitle": "Barista",
    "employeeTitleId": 1,
    "firstName": "John",
    "friendshipIds": Array [],
    "id": 1,
    "lastName": "Doe",
    "photoRef": "",
    "roles": 0,
    "wantShifts": 1,
  },
  "loggingIn": false,
}
`;

exports[`app sign in starts 1`] = `
Object {
  "loggingIn": true,
}
`;

exports[`app sign out is done 1`] = `
Object {
  "currentUser": undefined,
  "loggingIn": false,
}
`;
